kornia/augmentation/random_generator/random_generator.py

```diff
@@ -354,6 +354,9 @@ def random_crop_size_generator(
     _joint_range_check(scale, "scale")
     _joint_range_check(ratio, "ratio")
 
+    if batch_size == 0:
+        return dict(size=torch.zeros([0, 2]))
```

Comment:
Why returning [0, 2]? Maybe returning an empty tensor?

It should not be running according to:

https://github.com/kornia/kornia/blob/master/kornia/augmentation/base.py#L161-L163

Reply:
It's gonna be an empty tensor.
Right, but the parameter generation will earlier and https://github.com/kornia/kornia/blob/fc4b5a452110478e67de3f0a257e18ed036e18e7/kornia/augmentation/random_generator/random_generator.py#L370-L371 will crash.

Reply:
I see. I do not think returning random stuff is appropriate here.

How about something like:
```python
if batch_size != 0:
   w_out = w[torch.arange(0, batch_size), torch.argmax(cond, dim=1)] 
   h_out = h[torch.arange(0, batch_size), torch.argmax(cond, dim=1)] 
else:
   w_out = w
   h_out = hd
```

The idea is we should always output something meaningful to users.

Reply:
It's not really random stuff, since the function returns a tensor with shape [batch_size, 2], so returning one with shape of [0,2] for the `batch_size==0` case fits the pattern.